cirq/google/op_deserializer.py

```diff
@@ -136,3 +136,9 @@ def _args_from_proto(self, proto: v2.program_pb2.Operation
             if value is not None:
                 return_args[arg.constructor_arg_name] = value
         return return_args
+
+    def __eq__(self, other):
```

Comment:
Is this type also hashable? Implement `__hash__` if so. Can use `value.value_equality` for convenience.

Reply:
On further reflection, I don't think my equality function is very good, and I don't think these should generally be comparable, since they can include callable, so I moved the value equality function to the tests and got rid of it here.